### ISSUE TITLE
Unused var isInitializedLocalMap_ warning in Xpetra_EpetraMap.hpp triggering build error in SPARC Trilinos Integration builds with -Werror

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#4891]
CC: @trilinos/xpetra , @kddevin (Trilinos Data Services Product Lead), @bartlettroscoe, @fryeguy52

<Checklist>
<???: Add label "client: ATDM">
<???: Add label "ATDM Sev: Blocker" (by default but could be other "ATDM Sev: XXX")>
<???: Add label "type: bug"?>
<???: Add label for affected packages (e.g. "pkg: MueLu", "pkg: Tpetra", "pkg: Kokkos", etc.)>
<???: Add label "PA: ???Project Area???" (e.g. "PA: Linear Solvers", "PA: Data Services")>
<???: Add milestone "Initial cleanup of new ATDM ..." or "Keep promoted ATDM ...">
<???: Once GitHub Issue is created, add entries for tests to TrilinosATDMStatus/*.csv files>

## Next Action Status

<status-and-or-first-action>

## Description

The new unused var warning shown, for example, [here](https://testing.sandia.gov/cdash-dev-view/viewBuildError.php?type=1&buildid=4878569) in the build:

* `Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt`

in the build of MueLu which shows the warning:

```
[CTest: warning matched] /scratch/rabartl/Trilinos.base/NightlyBuilds/Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt/SRC_AND_BUILD/Trilinos/packages/xpetra/src/Map/Xpetra_EpetraMap.hpp:699:18: warning: private field 'isInitializedLocalMap_' is not used [-Wunused-private-field]
    mutable bool isInitializedLocalMap_ = false; // It's OK to use C++11 when Tpetra is enabled
```

is triggering build errors for the `clang-5.0.1` and `gnu-7.2.0` SPARC Trilinos Integration builds that have `-Werror` set.  One can see these build errors, for example, [here](http://compsim-dashboard.sandia.gov/cdash/viewBuildError.php?buildid=107482).

## Current Status on CDash

The status of these warnings can be seen in:

* [MueLu warning in Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt build over last 5 days](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&date=2019-04-13&filtercount=3&showfilters=1&filtercombine=and&field1=subprojects&compare1=93&value1=MueLu&field2=buildname&compare2=61&value2=Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt&field3=buildstarttime&compare3=83&value3=5%20days%20ago).

Just click on the number (e.g. '5') under "Build | Warn" to see the warnings.

## Steps to Reproduce

One should be able to reproduce this failure on any CEE LAN RHEL6 machine as described in:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

More specifically, the commands given for the system <supported-atdm-system> are provided at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#cee-rhel6-environment
The exact commands to reproduce this issue should be:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh \
    Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt 

$ cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_MueLu=ON \
 $TRILINOS_DIR

$ make NP=16
```

and one should see the warnings.

**Comment by bartlettroscoe** [comment]
FYI: I marked this as `ATDM Sev: Critical` because it brings down all SPARC testing for this build and therefore results in hundreds of failing SPARC tests.  This essentially turns off all SPARC Trilinos Integration testing for the clang-5.0.1 build.

**Comment by jhux2** [comment]
@bartlettroscoe PR #4892 should fix this.

**Comment by bartlettroscoe** [comment]
This has been fixed for some time by my PR #4892 and we can see a 100% passing clang-5.0.1 build and tests for SPARC yesterday [here](http://compsim-dashboard.sandia.gov/cdash/index.php?project=SPARC&date=2019-04-17&filtercount=1&showfilters=1&field1=buildname&compare1=66&value1=-trildev).  (NOTE: The build errors in the Intel build are due to MeuLu lib problems described in #4929 and are unrelated to this.)
